### ISSUE TITLE
Return resfresh_token base on configuration for subsequent calls

### ISSUE THREAD

**Issue by eschizoid** [spring-projects/spring-security-oauth#658]
Currently, the `refresh_token` is only returned during the first call to `/oauth/token` or when the access token is expire.

I would like to know if there is a way to override this behavior (I understand the security risks of always returning the `refresh_token`) and always return it base on some configuration.

FYI
Google OAuth2 implementation allows this behavior.

**Comment by eschizoid** [comment]
@dsyer is this not feasible at all?

**Comment by dsyer** [comment]
I expect it is eminently feasible. A good candidate for a pull request probably.

**Comment by jvz** [comment]
We implemented this as an aspect for a workaround, but now we'll work on a real feature pull request. I signed the ICLA earlier today.

**Comment by jvz** [comment]
I'd actually like a little bit of feedback before diving into this. Should this feature be disabled by default? Also, should adding a refresh token force a re-authorization by the user? I think that's what Google Oauth2 does.

**Comment by dsyer** [comment]
I think we should disable by default (for backwards compatibility if nothing else). How would you force a re-authorization in the token endpoint, though? It's not a user facing endpoint.

**Comment by jvz** [comment]
Yeah, I wasn't sure about that either. The idea behind it was that if the app loses the refresh token, they have to ask the user for permission again. I think that's a more secure way to handle it, but I've never used Google OAuth2, so I'm going off of documentation here (I've only used this project for oauth2).

If it should be configurable, do you have any advice on how to approach that in a framework way? Normally I'd probably just us a property placeholder, but that seems hacky for a framework feature (even though some frameworks abuse properties a lot).
